src/features/filter.ts

```diff
@@ -1,10 +1,39 @@
-export const actions = { /* put action creators here */};
+import { Todo } from '../types/Todo';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type PayloadType = { todos: Todo[]; query: string };
+type AllFilter = { type: 'filter/ALL'; payload: PayloadType };
+type ActiveFilter = { type: 'filter/ACTIVE'; payload: PayloadType };
+type CompletedFilter = { type: 'filter/COMPLETED'; payload: PayloadType };
+
+const filterAll = (todos: Todo[], query = '')
+: AllFilter => ({ type: 'filter/ALL', payload: { todos, query } });
+const filterActive = (todos: Todo[], query = '')
+: ActiveFilter => ({ type: 'filter/ACTIVE', payload: { todos, query } });
+const filterCompleted = (todos: Todo[], query = '')
+: CompletedFilter => ({ type: 'filter/COMPLETED', payload: { todos, query } });
+
+export const actions = { filterAll, filterActive, filterCompleted };
+
+type State = Todo[] | [];
+type Actions = AllFilter | ActiveFilter | CompletedFilter;
+
+export const filterReducer = (state: State = [], action: Actions) => {
+  switch (action.type) {
+    case 'filter/ALL':
+      return action.payload.todos
+        .filter(item => item.title.includes(action.payload.query));
```

Comment:
just set selected state type here instead of filtering todos
you don't need 3 different actions to handle status - create one `'filter/STATUS'` to save status provided in payload